File: qps/httpServer.go
package qps

import (
	"fmt"
	"io"
	"net/http"
	"time"
)

// wrk 测试结果
// wrk -c 10 -d 10s http://localhost:9000/hello

//   2 threads and 10 connections
//   Thread Stats   Avg      Stdev     Max   +/- Stdev
//     Latency     4.89ms   15.68ms 208.76ms   93.54%
//     Req/Sec    10.16k     6.11k   24.37k    62.37%

//   201100 requests in 10.10s, 24.55MB read
//   Requests/sec:  19906.57
//   Transfer/sec:      2.43MB

func QpsApp() {
	http.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {

		go func() {
			fmt.Println("127.0.0.1 - -  [" + time.Now().Format("2006-01-02 15:04:05") + "]  \"GET /hello HTTP/1.1\" 200")
		}()

		io.WriteString(w, "hello world")
	})
	http.ListenAndServe(":9000", nil)
}

File: qps/http_server.py
from http.server import HTTPServer,SimpleHTTPRequestHandler
from http import HTTPStatus
import socketserver

# wrk 测试结果
# wrk -c 10 -d 10s http://localhost:8000/hello
#   2 threads and 10 connections

#   Thread Stats   Avg      Stdev     Max   +/- Stdev
#     Latency     2.08ms    2.03ms  18.95ms   85.90%
#     Req/Sec   255.34    194.74     1.60k    83.16%

#   5042 requests in 10.03s, 635.30KB read

#   Socket errors: connect 0, read 5093, write 13, timeout 0
# Requests/sec:    502.72
# Transfer/sec:     63.34KB

class MyHTTPRequestHandler(SimpleHTTPRequestHandler):
    def do_GET(self):
        path = str(self.path)
        if(path == "/hello"):
            self.send_response(HTTPStatus.OK)
            self.send_header("Content-type","text/html")
            self.end_headers()
            self.wfile.write("hello world".encode())

def run(server_class=HTTPServer, handler_class=MyHTTPRequestHandler):
    server_address = ('', 8000)
    httpd = server_class(server_address, handler_class)
    httpd.serve_forever()

if __name__ == "__main__":
    run()